aptos-move/framework/aptos-experimental/sources/trading/market/market_types.move

```diff
@@ -111,14 +112,14 @@ module aptos_experimental::market_types {
         V1 {
```

Comment:
can you explain the semantics here exactly, i.e. in a path/state machine of order placement, which functions will be called when?

i.e. is only one of the validate_order_placement_f vs place_maker_order_f called for every order, based on whether it is a taker or maker, or is validate_order_placement_f always called? 
is cleanup_order_f called for every order that either of them was called, or just on those that weren't settled? 


based on that I might have more comments, but it's hard to check the logic without fully understanding that.

Reply:
`validate_order_placement_f` is always called irrespective of if its taker or maker
`place_maker_order_f` is called just before the order is inserted into the orderbook
`cleanup_order_f` is always called irrespective of if its taker or maker.

Reply:
One more is for every order there is one and only one cleanup_order_f is called and the size is the size of the order being cleaned up - it can be 0, if the order was fully matched